tortoise/pydantic.py

```diff
@@ -0,0 +1,384 @@
+import functools
+import inspect
+import re
+import typing
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type
+
+import tortoise
+from tortoise import fields, models
+
+if typing.TYPE_CHECKING:
+    from tortoise.models import Model
+
+
+# Pydantic is an optional dependency
+try:
+    import pydantic
+    from pydantic import BaseModel
+
+    @functools.lru_cache()
+    def _get_comments(cls: Type["Model"]) -> Dict[str, str]:
+        """
+        Get comments exactly before attributes
+
+        It can be multiline comment. The placeholder "{model}" will be replaced with the name of the
+        model class.
+
+        :param cls: The class we need to extract comments from its source.
+        :return: The dictionary of comments by field name
+        """
+        source = inspect.getsource(cls)
+        comments = {}
+
+        for cls in reversed(cls.__mro__):
+            if cls is object:
+                continue
+            matches = re.findall(rf"((?:(?!\n|^)[^\w\n]*#.*?\n)+?)[^\w\n]*(\w+)\s*[:=]", source)
+            for match in matches:
+                field_name = match[1]
+                # Extract text
+                comment = re.sub(r"(^\s*#\s*|\s*$)", "", match[0], flags=re.MULTILINE)
+                # Class name template
+                comment = comment.replace("{model}", cls.__name__)
+                # Change multiline texts to HTML
+                comments[field_name] = comment.replace("\n", "<br>")
+
+        return comments
+
+    @functools.lru_cache()
+    def _get_annotations(cls: Type["Model"], method: Optional[Callable] = None) -> Dict[str, Any]:
```

Comment:
Ditto, this should be standard.

Reply:
It is good because it can handle the `ForwardRef` annotations like this: `ForeignKeyRelation['Address']`.

Reply:
`ForwardRef` caused me lots of pain with type annotation in docgen. Sometimes the ref isn't in scope because it is hidden behind a `if TYPE_CHECKING:` condition :'(